portfolio/src/main/java/com/google/sps/servlets/BasketballCourtsServlet.java

```diff
@@ -0,0 +1,23 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.BasketballCourts;
+
+/** Returns the accessible basketball courts in New York City */
+@WebServlet("/basketball-courts")
+public class BasketballCourtsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    List<BasketballCourts.BasketballCourt> accessibleCourts = BasketballCourts.getAccessibleBasketballCourts();
```

Comment:
yeah that's what i was worried about: every GET request will lead to a file read. you definitely want to cache your List in your BasktballCourts. Also rename BasketballCourts to BasketballCourtsDatastore. it's the same as a db except it uses the file as source of data and has no mutation.